src/NUnitConsole/nunit3-console/ResultReporter.cs

```diff
@@ -42,6 +42,8 @@ public ResultReporter(XmlNode resultNode, ExtendedTextWriter writer, ConsoleOpti
             OverallResult = resultNode.GetAttribute("result");
             if (OverallResult == "Skipped")
                 OverallResult = "Warning";
+            if (OverallResult == null)
+                OverallResult = "Unknown";
```

Comment:
Not really. Exceptions in other threads must be handled properly. Right now even  logging

string stopError = string.Format("Failed to stop the remote agent. {0}", e.Message);

in the ProcessRunner.cs does not write into log anything here.

In my second commit I've added output coloring as Error though.